.github/workflows/test.yml

```
@@ -23,48 +36,48 @@ jobs:
       - name: Checkout the repo
         uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
 
-      # Workaround to provide additional free space for testing.
-      # Reference: https://github.com/actions/runner-images/issues/2840
-      - name: Free disk space
-        if: startsWith(runner.name, 'GitHub Actions') && startsWith(matrix.os, 'ubuntu')
-        run: sudo rm -rf /usr/share/dotnet
-
       - name: Install Rust toolchain
         run: |
           rustup toolchain install --no-self-update stable --profile minimal -c clippy
           rustup default stable
 
-      - name: Install Rust nightly toolchain
-        if: github.event_name != 'schedule'
-        run: rustup toolchain install --no-self-update nightly --profile minimal -c rustfmt
-
-      - name: Format check
-        if: github.event_name != 'schedule'
-        run: cargo +nightly fmt --all -- --check
-
       - name: Build
         run: cargo build --locked
 
       - name: Clippy
         if: github.event_name != 'schedule'
         run: cargo clippy --locked -- -D warnings
```

```suggestion
        run: cargo lint
```
Not exactly what this PR is looking to change...but an easy change. The suggestion is to use the form that is more thorough and also more likely to be used locally. From the `.cargo/config` file in this repository:

`lint = "clippy --locked --all-features --all-targets -- -D warnings"`

---

Just adding it to `.cargo/config` doesn't make it more likely to be used locally... It just makes it more confusing by making us look elsewhere to figure out what's actually being run...

As for `--all-features --all-targets` , that may require code changes and is beyond the scope of this PR.

---

> Just adding it to .cargo/config doesn't make it more likely to be used locally... It just makes it more confusing by making us look elsewhere to figure out what's actually being run...

100% agree, I've also never used `cargo lint` and I've said it in other PRs that I'd rather get rid of that than change this CI step to use it.

> As for --all-features --all-targets , that may require code changes and is beyond the scope of this PR.

That is fair though the changes are absolutely minimal so it should be fine tbh.